sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala

```diff
@@ -132,13 +132,12 @@ case class GetJsonObject(json: Expression, path: Expression)
     }
 
     if (parsed.isDefined) {
+      val parser = jsonFactory.createParser(jsonStr.getBytes)
+      val output = new ByteArrayOutputStream()
+      val generator = jsonFactory.createGenerator(output, JsonEncoding.UTF8)
```

Comment:
If those are also objects that need to be closed, I agree. Otherwise if those lines fail, there is no generator to close. Moving this outside the try does change things but it looks like the intent since the catch is only for an error during parsing. I think.